### PR TITLE
docker-buildx: rename from moby-buildx and upgrade to 0.12.1

### PR DESCRIPTION
**Pull request by hbeberman**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Rename moby-buildx to docker-buildx and upgrade to version 0.12.1

###### Change Log  

- Rename moby-buildx to docker-buildx
- Upgrade docker-buildx to version 0.12.1
- Clean up spec slightly

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local package build
- Buddy build https://dev.azure.com/mariner-org/mariner/_build/results?buildId=515256&view=results

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -26839,6 +26829,16 @@
         }
       }
     },
+    {
```
Comment (ms-mahuber) [inline cgmanifest.json:26832]: just curious, was this intentionally moved down or was this done by tooling?

File: SPECS/docker-buildx/docker-buildx.spec [hunk 2]
```diff
@@ -34,14 +36,18 @@ go build -mod=vendor \
...
+mkdir -p "%{buildroot}%{_libexecdir}/docker/cli-plugins"
+install -m 755 buildx "%{buildroot}%{_libexecdir}/docker/cli-plugins/docker-buildx"
 
 %files
 %license LICENSE
 %{_libexecdir}/docker/cli-plugins/docker-buildx
 
 %changelog
+* Tue Feb 27 2024 Henry Beberman <henry.beberman@microsoft.com> - 0.12.1-1
+- Rename package frmo moby-buildx to docker-buildx
```
Comment (ms-mahuber) [inline SPECS/docker-buildx/docker-buildx.spec:48]: frmo

File: cgmanifest.json (hunk 1 above)
Comment (hbeberman) [reply cgmanifest.json:26832]: This was me intentionally moving it down. Missed the re-alphabetization when renaming from moby-runc in a previous PR.

File: SPECS/docker-buildx/docker-buildx.spec (hunk 2 above)
Comment (hbeberman) [reply SPECS/docker-buildx/docker-buildx.spec:48]: 🤦

**Review by ms-mahuber — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**